sklearn/cluster/k_means_.py

```diff
@@ -293,6 +308,15 @@ def k_means(X, n_clusters, init='k-means++', precompute_distances='auto',
     if _num_samples(X) < n_clusters:
         raise ValueError("n_samples=%d should be >= n_clusters=%d" % (
             _num_samples(X), n_clusters))
+
+    # set sample_weights if None passed
+    sample_weights = _check_sample_weights(X, sample_weights)
+
+    # verify that the number of samples is equal to the number of weights
+    if _num_samples(X) != len(sample_weights):
```

Comment:
You can also use `sklearn.utils.validation.check_consistent_length`.
Also, this check does not seem to be tested.

Reply:
I could use `sklearn.utils.validation.check_consistent_length`, but then the error message will be very generic. Would you still prefer this?

I'll add tests for the check method.

Reply:
Right, I have no strong feeling about it, you can keep it this way.